woocommerce/google-listings-and-ads
# Tree Select Components - Expand/collapse

### Changes proposed in this Pull Request:

Adding functionality for expanding and collapsing groups of nodes in the tree. 

Thread info: pcTzPl-fg-p2

### Screenshots:

https://user-images.githubusercontent.com/5908855/156535643-e6331260-db12-4ee6-a000-2303a86fba58.mov

### Detailed test instructions:

1. Checkout the branch
2. Go to `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fcomponent-test` 
3. You can see component TreeSlectControl with the label `Select Country` and the placeholder `Select`
4. Clicking on "Select" expands the TreeSlector and shows text cursor instead of the placeholder  
5. Tree Shows  the groups collapsed 
6. Clicking on one group expands the group but not the subgroups
7. Clicking again close the group. 

### Additional details:

- Implemented expand/collapse logic
- Updated Aria Roles to use Tree instead of Listbox
- Instead of `list.js` file with `List` and `Options` component inside it. We renamed it to `Options` and located the wrapper directly in the main component file. 
- Updated CSS [according to this comment](https://github.com/woocommerce/google-listings-and-ads/pull/1243#issuecomment-1054538974). Please notice in Figma the boxes are "aligned" as in the previous version. Maybe good to discuss this with the Design team. 
- Refactor Tests dividing them into multiple files  
- It's pending an extra PR in the future with alternative ways to handle option changes, search, expand/collapse etc...
